tests/testthat/test-differentialAbundance.R

```diff
@@ -431,3 +431,74 @@ test_that("differentialAbundance catches deseq errors", {
 
 
 })
+
+test_that("differentialAbundance method Maaslin does stuff",{
+    df <- testOTU
+  counts <- round(df[, -c("entity.SampleID")]*1000)
+  counts[ ,entity.SampleID:= df$entity.SampleID]
+  nSamples <- dim(df)[1]
+  testSampleMetadata <- data.frame(list(
+    "entity.SampleID" = df[["entity.SampleID"]],
+    "entity.binA" = rep(c("binA_a", "binA_b"), nSamples/2, replace=T),
+    "entity.cat3" = rep(paste0("cat3_", letters[1:3]), nSamples/3, replace=T),
+    "entity.cat4" = rep(paste0("cat4_", letters[1:4]), nSamples/4, replace=T),
+    "entity.contA" = rnorm(nSamples, sd=5)
+    ))
+
+
+  testCountsData <- microbiomeComputations::AbsoluteAbundanceData(
+              data = counts,
+              sampleMetadata = testSampleMetadata,
+              recordIdColumn = 'entity.SampleID')
+
+  tetData <- microbiomeComputations::AbundanceData(
+    data = df,
+    sampleMetadata = testSampleMetadata,
+    recordIdColumn = 'entity.SampleID'
+  )
+
+  comparatorVariable <- microbiomeComputations::Comparator(
+                          variable = veupathUtils::VariableMetadata(
+                            variableSpec = VariableSpec(
+                              variableId = 'cat4',
+                              entityId = 'entity'
+                            ),
+                            dataShape = veupathUtils::DataShape(value="CATEGORICAL")
+                          ),
+                          groupA = veupathUtils::BinList(
+                            S4Vectors::SimpleList(
+                              c(veupathUtils::Bin(
+                                binLabel="cat4_a"
+                              ))
+                            )
+                          ),
+                          groupB = veupathUtils::BinList(
+                            S4Vectors::SimpleList(
+                              c(veupathUtils::Bin(
+                                binLabel="cat4_b"
+                              ))
+                            )
+                          )
+  )
+
+  result <- differentialAbundance(testData,
+              comparator = comparatorVariable,
+              method='Maaslin',
+              verbose=F)
+  dt <- result@data
+  stats <- result@statistics
+
+
+  resultCounts <- differentialAbundance(testCountsData,
+              comparator = comparatorVariable,
+              method='Maaslin',
+              verbose=F)
+  dtCounts <- result@data
+  statsCounts <- result@statistics
+
+  expect_equal(dt, dtCounts)
+  #kind of naively figuring the one is more sensitive/ specific than the other
+  expect_true(length(stats$pointID) > 0)
```

Comment:
does `length(stats$pointID)` change each time we run it? I would have guessed no but maybe maaslin is doing something stochastic in its modelling